### PR TITLE
Additional libraries and indices in sample submission template.

### PR DESCRIPTION
**Pull request by UlysseFG**
- [x] Add new library to sample submission template.
- [x] Create many to many link between index and index set and import existing relations.
- [x] Fix the index endpoint and frontend index list and detail to support multiple sets by index.
- [x] Make Index creation template import use a get_or_create_index service to reuse existing index and tie them to new sets.
- [x] Rename index sets to use clarity naming scheme.
- [x] Add index set present in the sample submission template to index that were created without sets.
- [x] Create new index / index set that did not exist (include sets that are a mix of old and new indices).

### CODE REVIEW

File: backend/fms_core/migrations/0066_v4_12_0.py
Comment (nafiz1001) [inline backend/fms_core/migrations/0066_v4_12_0.py:1]: One of the few cases where having all elements of an array in the same line would make the review process easier ;)

File: backend/fms_core/migrations/0066_v4_12_0.py [hunk 1]
```diff
@@ -0,0 +1,7090 @@
...
+                                                 "SI_TT7_H04-SI_TT5_H04",
+                                                 "SI_TT7_H05-SI_TT5_H05",
+                                                 "SI_TT7_H06-SI_TT5_H06",
+                                                 "SI_TT7_H07-SI_TT5_H07",
+                                                 "SI_TT7_H08-SI_TT5_H08",
+                                                 "SI_TT7_H09-SI_TT5_H09",
+                                                 "SI_TT7_H10-SI_TT5_H10",
+                                                 "SI_TT7_H11-SI_TT5_H11",
+                                                 "SI_TT7_H12-SI_TT5_H12"],
+        "Agilent-SureSelect_XT_HS_high-sensitivity":["SSXTHS701",
```
Comment (nafiz1001) [inline backend/fms_core/migrations/0066_v4_12_0.py:247]: (optional) space after colon

File: backend/fms_core/migrations/0066_v4_12_0.py [hunk 2]
```diff
@@ -0,0 +1,7099 @@
...
+                                                     "SSXTHS724",
+                                                     "SSXTHS725",
+                                                     "SSXTHS726",
+                                                     "SSXTHS727",
+                                                     "SSXTHS728",
+                                                     "SSXTHS729",
+                                                     "SSXTHS730",
+                                                     "SSXTHS731",
+                                                     "SSXTHS732"],
+        "Broad_Institute-sgRNA_guide_RNA_index_definition":["BROAD_sgRNA_A01",
```
Comment (nafiz1001) [inline backend/fms_core/migrations/0066_v4_12_0.py:279]: (optional) space after colon

File: frontend/src/models/fms_api_models.ts [hunk 3]
```diff
@@ -163,7 +163,7 @@ export interface FMSImportedFile {
 
 export interface FMSIndex extends FMSTrackedModel {
     name: string                        // eg "Index_1"
-    index_set: string                   // IndexSet name
```
Comment (nafiz1001) [inline frontend/src/models/fms_api_models.ts:166]: `string[]`

File: frontend/src/models/fms_api_models.ts [hunk 4]
```diff
@@ -163,7 +163,7 @@ export interface FMSImportedFile {
...
+    index_sets: string                   // IndexSet name
```
Comment (nafiz1001) [inline frontend/src/models/fms_api_models.ts:166]: `string[]`

**Comment by nafiz1001** [comment]
```
  Applying fms_core.0066_v4_12_0...Traceback (most recent call last):
  File "/home/nafiz/projects/freezeman/backend/manage.py", line 21, in <module>
    main()
  File "/home/nafiz/projects/freezeman/backend/manage.py", line 17, in main
    execute_from_command_line(sys.argv)
  File "/home/nafiz/projects/freezeman/backend/env/lib/python3.11/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
    utility.execute()
  File "/home/nafiz/projects/freezeman/backend/env/lib/python3.11/site-packages/django/core/management/__init__.py", line 436, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/nafiz/projects/freezeman/backend/env/lib/python3.11/site-packages/django/core/management/base.py", line 412, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/home/nafiz/projects/freezeman/backend/env/lib/python3.11/site-packages/django/core/management/base.py", line 458, in execute
    output = self.handle(*args, **options)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nafiz/projects/freezeman/backend/env/lib/python3.11/site-packages/django/core/management/base.py", line 106, in wrapper
    res = handle_func(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nafiz/projects/freezeman/backend/env/lib/python3.11/site-packages/django/core/management/commands/migrate.py", line 356, in handle
    post_migrate_state = executor.migrate(
                         ^^^^^^^^^^^^^^^^^
  File "/home/nafiz/projects/freezeman/backend/env/lib/python3.11/site-packages/django/db/migrations/executor.py", line 135, in migrate
    state = self._migrate_all_forwards(
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nafiz/projects/freezeman/backend/env/lib/python3.11/site-packages/django/db/migrations/executor.py", line 167, in _migrate_all_forwards
    state = self.apply_migration(
            ^^^^^^^^^^^^^^^^^^^^^
  File "/home/nafiz/projects/freezeman/backend/env/lib/python3.11/site-packages/django/db/migrations/executor.py", line 252, in apply_migration
    state = migration.apply(state, schema_editor)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nafiz/projects/freezeman/backend/env/lib/python3.11/site-packages/django/db/migrations/migration.py", line 132, in apply
    operation.database_forwards(
  File "/home/nafiz/projects/freezeman/backend/env/lib/python3.11/site-packages/django/db/migrations/operations/special.py", line 193, in database_forwards
    self.code(from_state.apps, schema_editor)
  File "/home/nafiz/projects/freezeman/backend/fms_core/migrations/0066_v4_12_0.py", line 6969, in delete_ficticious_index
    index_by_set = IndexBySet.objects.get(index__name=index_name)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nafiz/projects/freezeman/backend/env/lib/python3.11/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nafiz/projects/freezeman/backend/env/lib/python3.11/site-packages/django/db/models/query.py", line 637, in get
    raise self.model.DoesNotExist(
__fake__.IndexBySet.DoesNotExist: IndexBySet matching query does not exist.
```

**Comment by UlysseFG** [comment]
you need a more recent backup ... i create new index and index sets on the prod server using templates but the last DB i posted did not have those ... if you want i can provide an updated DB ...

**Comment by nafiz1001** [comment]
> you need a more recent backup ... i create new index and index sets on the prod server using templates but the last DB i posted did not have those ... if you want i can provide an updated DB ...

After this PR gets merged 🙂

**Comment by nafiz1001** [comment]
Looks like you gotta rebase ;)

File: frontend/src/models/fms_api_models.ts [hunk 5]
```diff
@@ -163,7 +163,7 @@ export interface FMSImportedFile {
 
 export interface FMSIndex extends FMSTrackedModel {
     name: string                        // eg "Index_1"
-    index_sets: string                   // IndexSet name
+    index_sets: string[]                // IndexSet name
```
Comment (nafiz1001) [inline frontend/src/models/fms_api_models.ts:166]: `IndexSet names` ;)

**Pull request merged**